src/Microsoft.AspNet.Authentication.OAuth/Events/IOAuthEvents.cs

```diff
@@ -16,19 +16,19 @@ public interface IOAuthEvents
         /// </summary>
         /// <param name="context">Contains information about the login session.</param>
         /// <returns>A <see cref="Task"/> representing the completed operation.</returns>
-        Task Authenticated(OAuthAuthenticatedContext context);
+        Task CreatingTicket(OAuthCreatingTicketContext context);
 
         /// <summary>
         /// Invoked prior to the <see cref="ClaimsIdentity"/> being saved in a local cookie and the browser being redirected to the originally requested URL.
         /// </summary>
         /// <param name="context"></param>
         /// <returns>A <see cref="Task"/> representing the completed operation.</returns>
-        Task ReturnEndpoint(OAuthReturnEndpointContext context);
+        Task SigningIn(SigningInContext context);
 
         /// <summary>
         /// Called when a Challenge causes a redirect to the authorize endpoint.
         /// </summary>
         /// <param name="context">Contains redirect URI and <see cref="AuthenticationProperties"/> of the challenge.</param>
-        Task ApplyRedirect(OAuthApplyRedirectContext context);
+        Task RedirectToAuthorizationEndpoint(OAuthRedirectToAuthorizationContext context);
```

Comment:
I was leaning towards going through and removing Endpoint on all of the methods/API except for the Options, since it doesn't appear to add that much meaning and that name is HUMONGOUS.